src/lib/dataset.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * This module contains helpers and definitions related to datasets.
+ */
+
+export type DataSetColumn = {
+  name: string;
+  type?: 'integer' | 'float' | 'boolean' | 'string' | 'date' | 'object';
+};
+
+export type DataSet = {
+  columns: Array<DataSetColumn>;
+  data: Array<Array<any>>;
```

Comment:
Maybe an example dataset would be nice here. We have to understand at once what is the orientation of data (it is an arrays of rows, not columns)